### ISSUE TITLE
ui.use_standins_for_native_colors

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#129]
```js
/* 2609: disable exposure of system colors to CSS or canvas (FF44+)
 * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=232227
 * [2] https://bugzilla.mozilla.org/show_bug.cgi?id=1330876 ***/
user_pref("ui.use_standins_for_native_colors", true); // (hidden pref)
```

Note: the original ticket (14 years ago!?!) was about contrasting colors in high contrast theme in XP but morphed IMO - 11 years later it became a patch to not expose system colors (the code module still uses XP in its name). I think this still happens (the black on black type issue, not the leaking of system colors) - see comments in upcoming ticket [1330876](https://bugzilla.mozilla.org/show_bug.cgi?id=1330876) - this ticket is in the Tor Uplift project.

I'm keen to add this and then let everyone complain about black on black or whatever :) and then I can say there is a fix coming and to not worry about it and just suck it up :)

**Comment by earthlng** [comment]
> Since the routine affected by the preference "ui.use_standins_for_native_colors" does not include nsLayoutUtils::GetColor, how can the patch fix the bug?

Since not even Arthur seems to understand what exactly the Tor patch fixes and neither does the mozilla dev, I'd say let's wait and see how the ticket evolves before we rush to include this

**Comment by Thorin-Oakenpants** [comment]
The preference (introduced in mozilla 232227) disables exposure of system colors. That works and has done since FF44. It has one particular bug with a particular color set, not a bug in blocking leakage of system colors: quote Arthur: "but that mode doesn't operate quite correctly when the desktop theme calls for white on black text". It's not leaking system colors and is good to go.

The second ticket will fix that one minor "display" bug. The fact that Chung-Sheng isn't sure what the tor patch does, nor Arthur who didn't work on it (Kathy did), or if the tor patch even works in tor, is irrelevant. The second ticket is for FF to build it's own patch and will be resolved for FF (and uplifted to tor). Probably by 55 (seems like a hard deadline for a lot of things is 55)

OT: FYI see [1367313](https://bugzilla.mozilla.org/show_bug.cgi?id=1367313) which leads to -> [1344669](https://bugzilla.mozilla.org/show_bug.cgi?id=1344669) = bye bye pref 2410, because some one said oh lets remove it .. godamnit

**Comment by atomGit** [comment]
`ui.use_standins_for_native_colors` makes tool-tips unreadable for me - the font color seems to be same as the background color

[from post above...](https://github.com/ghacksuserjs/ghacks-user.js/issues/129#issuecomment-303812609)

> It has one particular bug with a particular color set, not a bug in blocking leakage of system colors: quote Arthur: "but that mode doesn't operate quite correctly when the desktop theme calls for white on black text".

**Comment by Thorin-Oakenpants** [comment]
It's a FF bug, and will be addressed. The original pref and results still stand. I'm not going to try and work out exactly WTF is going on. It's to do with fields with no color specified, so it falls back to set colors (not system ones) and the fallback color (the hard-coded set ones in the patch) is always black text, and then there's something about special properties for accessibility in Windows, so it manifests a lot in High Contrast.

@atomGit What linux are you on, and what colors do you get? Is it black on black, or are you saying if the tooltip background is pink, you get pink text, and if it's red you get red?

**Comment by atomGit** [comment]
Linux Mint 18.1

tool-tips always seem to have a black background color (unless set differently by the document i suppose) and, with this pref = true, the text also appears black

**Comment by Thorin-Oakenpants** [comment]
Right .. so its the same issue as already discussed. It will get fixed when they close 1330876. Looks like a big push to get a lot of tor uplift items done for 55 as a bit of a milestone,. so fingers crossed. I have no issues on Windows, btw - and the pref exists for all OS'es, I guess Linux, Windows, Mac handle it all differently. Wonder what @nodiscc gets.

**Comment by Thorin-Oakenpants** [comment]
[1330876](https://bugzilla.mozilla.org/show_bug.cgi?id=1330876) resolved fixed for FF56
